### PR TITLE
Soapstones: Leave messages for the future crew

### PR DESCRIPTION
**Pull request by coiax**
:cl: coiax
add: The Librarian now starts with a chisel/soapstone/chalk/magic marker capable of engraving messages for subsequent shifts, and permanently erasing messages that the Librarian is unhappy with. It has limited uses, so order more at Cargo.
fix: The contraband cream pie crate is now locked, and requires Theatre access.
/:cl:

- Engraved messages can be manually smashed or exploded, but will still appear next shift until properly erased by a soapstone.
- An engraved message's colour is randomly determined based on the message's contents (ie. md5)
- Engravings cannot be done on shuttles, off station, and are map specific.

image

TODO
- [x] Actual persistence
- [x] Sprites

### CODE REVIEW

File: code/modules/library/lib_items.dm [hunk 1]
```diff
@@ -316,3 +316,134 @@
...
+	else if(!(isfloorturf(target) || iswallturf(target)))
+		. = FALSE
+	else
+		. = TRUE
+
+/obj/structure/chisel_message
+	name = "engraved message"
+	desc = "A message from a past traveler."
+	icon = 'icons/obj/stationobjs.dmi'
+	icon_state = "darksouls_messages"
```
Comment (Xhuis) [inline code/modules/library/lib_items.dm]: Very subtle

Comment (coiax) [inline code/modules/library/lib_items.dm]: I'm literally copy and pasting parts of goofballs code, blame him not me

**Comment by Xhuis** [comment]
Are messages written from a predetermined set of words? That was the fun part in the Soulsborne games - trying to make up meaningful stuff with what you had, or just being crude about it. A rudimentary rating system would also be cool but not important.

Also, only having the librarian start with this might be meh. Another way to get them so people could write more messages would be great.

Gripes aside this is cool and I like it.

**Comment by lzimann** [comment]
It's time to move AWAY from persistance, not add even more.

**Comment by KorPhaeron** [comment]
Why?

**Comment by coiax** [comment]
@KorPhaeron this is part of goof's super popular librarian archaeologist PR. I'm doing the easy part of soapstones, and then maybe trophy cases later.

**Comment by YotaXP** [comment]
Could this cause problems when messages like "Urist McTator tried to kill me!" show up in later shifts with a character of the same name?  I'd hate to have to add a whole section of rules for soapstone messages.

**Comment by XDTM** [comment]
I'd say pre-determined messages would prevent specific namecalling.

**Comment by Iamgoofball** [comment]
That's an amazing meta breaking tactic

**Comment by coiax** [comment]
Goofball, this is your feature! Jeez, everyone's a critic

**Comment by Xhuis** [comment]
You can take inspiration from DS itself for the grammar, as I can see right now that you can enter anything you want (which would kill the feature for me). For example:

```
try ****
time for ****
be wary of ****
****
****?
****!
visions of ****...
```

and

```
traitor
cultist
Syndicate
Nanotrasen
assistant

```

**Comment by Anonmare** [comment]
I'd like predetermined because I don't want some memelord to put the Navy Seals Copypasta or WGW on every tile in a hallway and make examining people running down a hallway a treacherous task.

**Comment by ShadowDeath6** [comment]
great chest ahead!

**Comment by Xhuis** [comment]
Amazing* you goddamn amateur

**Comment by ExcessiveUseOfCobblestone** [comment]
Put finger in hole.

Try Jumping!

**Comment by Iamgoofball** [comment]
@coiax i know, I was complimenting it

**Comment by Xhuis** [comment]
@ExcessiveUseOfCobblestone Goddamn you people's meme games aren't on point.
"try finger
but hole"

File: code/modules/library/soapstone.dm [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+	var/x = data["x"]
+	var/y = data["y"]
+	var/turf/newloc = locate(x, y, ZLEVEL_STATION)
+	forceMove(newloc)
+	update_icon()
+
+/obj/structure/chisel_message/examine(mob/user)
+	..()
+	user << "<span class='warning'>[hidden_message]</span>"
+	user << "This message was engraved by <span class='name'>[creator_name]</span>."
```
Comment (Xhuis) [inline code/modules/library/soapstone.dm]: \>messages say who wrote them
\>messages can be anything
\>messages have no rating system

MINIMUM TRIGGER ----------------------|- MAXIMUM TRIGGER

**Comment by Haevacht** [comment]
I regret making that orange soapstone icon slightly more now than when goof refused to make it just a chisel.

And if it HAS to be a fucking soapstone, predefined phrases, no iding the author (without fingerprints and shit).

**Comment by coiax** [comment]
Okay, I'll get rid of the author in the description.

**Comment by KingPhilipIII** [comment]
PRAISE THE SUN

**Comment by Kromgar** [comment]
Can lavaland get soapstones?


>Get to necropolis door 
>Amazing chest ahead

**Comment by Xhuis** [comment]
The potential here is immense.
In front of a monkey pen: "be wary of lizard"
In front of the CMO's office: "fatty ahead"

**Comment by coiax** [comment]
@MrStonedOne for some reason, the chisel messages saving proc isn't firing when I end the round, but it works if I call the proc manually. Can you see something I've done wrong?

**Comment by KorPhaeron** [comment]
I know it's part of goof's PR, I'm asking you what you think this will add to the game

**Comment by coiax** [comment]
Librarians will be able to leave messages, memes, and warnings for the crew, as well as commentary on various parts of the map. Since they can only be permanently erased by a librarian, you could find messages or jokes hidden deep within maint. We track the ckey of anyone leaving a message for admin purposes, so abusers get shut down (or job banned from librarian topkek). This'll make the station feel more "lived in", and remind players that they're part of a community that is more than just one round, while not actually changing the gameplay round to round.

File: code/modules/library/soapstone.dm [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+	icon = 'icons/obj/items.dmi'
+	icon_state = "soapstone"
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_TINY
+	var/tool_speed = 50
+
+/obj/item/soapstone/New()
+	. = ..()
+	name = pick("soapstone", "chisel", "chalk", "magic marker")
```
Comment (ChangelingRain) [inline code/modules/library/soapstone.dm]: the desc and fluff messages should vary based on the name

**Comment by jughu** [comment]
>craves the bee movie script on the hydronics walls

**Comment by GunHog** [comment]
>> craves the bee movie script on the hydronics walls

Why crave it when you can carve it yourself? :stuck_out_tongue_closed_eyes:

**Comment by coiax** [comment]
oh god, sprite conflicts

**Comment by AnturK** [comment]
Alright, let's give this a try. If this turns into WGW or meta messages only, i'll open the revert personally.

**Pull request merged**